src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataControllerTest.java

```diff
@@ -152,6 +217,379 @@ void shouldFetchEmptyUserTypes() {
+        List<ServiceResponse> services = List.of(ServiceResponse.builder().serviceCode("BFA1").service("avd").build());
+        searchResponse.setServices(services);
+
+        SearchStaffUserResponse searchResponse2 = SearchStaffUserResponse.builder()
+                .firstName("firstName")
+                .lastName("mastName")
+                .emailId("emailId")
+                .region("region")
+                .regionId(123)
+                .build();
+        searchResponse2.setServices(services);
+
+        searchReq = SearchRequest.builder()
+                .serviceCode("BFA1")
+                .build();
+
+        advancedSearchResponse = new ResponseEntity<>(Arrays.asList(searchResponse,searchResponse2), null,
+                HttpStatus.OK);
+        when(staffRefDataService.retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class)))
+                .thenReturn(advancedSearchResponse);
+        ResponseEntity<List<SearchStaffUserResponse>> response =
+                staffRefDataController.searchStaffProfile(1,2, searchReq);
+        assertNotNull(response);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        verify(staffRefDataService, times(1))
+                .retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class));
+
+        assertTrue(validateSearchUserProfileResponse(response,searchReq));
+
+    }
+
+
+
+    @ParameterizedTest
+    @ValueSource(strings = {"101633,231596","231596"})
+    void should_return_staffProfiles_search_with_location_status_code_200(String location) {
+        List<Location> baseLocations = List.of(Location.builder().baseLocationId(231596).locationName("avd").build());
+        searchResponse.setBaseLocations(baseLocations);
+
+        searchReq = SearchRequest.builder()
+                .location(location)
+                .build();
+
+        advancedSearchResponse = new ResponseEntity<>(List.of(searchResponse), null, HttpStatus.OK);
+        when(staffRefDataService.retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class)))
+                .thenReturn(advancedSearchResponse);
+        ResponseEntity<List<SearchStaffUserResponse>> response =
+                staffRefDataController.searchStaffProfile(1,2, searchReq);
+        assertNotNull(response);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        verify(staffRefDataService, times(1))
+                .retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class));
+
+        assertTrue(validateSearchUserProfileResponse(response,searchReq));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"abcde,fjkui","*__"})
+    void should_return_staffProfiles_search_with_invalid_input_location_status_code_400(String location) {
+        searchReq = SearchRequest.builder()
+                .location(location)
+                .build();
+
+        assertThrows(InvalidRequestException.class, () -> staffRefDataController
```

Comment:
can you check for expected error message

Reply:
changes done